### DISCUSSION TITLE
HEVPRambergOsgoodHardening

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#25612] [Q&A General]
Hello, 

I do not understand what HEVPRambergOsgoodHardening should be returning, is it the value of 'val' or a boolean value (return true) ?

```
 bool
HEVPRambergOsgoodHardening::computeValue(unsigned int qp, Real & val) const
{
  val = _sig0 * std::pow(_intvar[qp] / _peeq0 + 1.0, _exponent);
  return true;
}

```

**Comment by GiudGiud** [comment]
Hello

that s an interesting function definition but it s actually both!

It returns true as you can see
But then it takes a reference to a real in its arguments and sets it to a value. And given the name of the function I imagine that it s its primary role.

please have a look at the code calling that function to see what the reference is equal to

Guillaume
